controllers/edgeconfig_controller.go

```diff
@@ -0,0 +1,240 @@
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *EdgeConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.V(1).Info("EdgeConfig Reconcile", "edgeConfig", req)
+
+	edgeConfig, err := r.EdgeConfigRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	if edgeConfig.DeletionTimestamp != nil {
+		return ctrl.Result{}, nil
+	}
+
+	edgeDevices, err := r.EdgeDeviceRepository.ListForEdgeConfig(ctx, edgeConfig.Name, edgeConfig.Namespace)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return ctrl.Result{Requeue: true}, err
+		}
+	}
+	err = r.addPlaybookExecutionToDevices(ctx, edgeConfig, edgeDevices)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *EdgeConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		// Uncomment the following line adding a pointer to an instance of the controlled resource as an argument
+		For(&v1alpha1.EdgeConfig{}).
+		WithEventFilter(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				return true
+			},
+		}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: r.MaxConcurrentReconciles}).
+		Complete(r)
+}
+
+func createPlaybookExecutions(edgeConfig *v1alpha1.EdgeConfig) []v1alpha1.PlaybookExecution {
+	playbookExecutions := []v1alpha1.PlaybookExecution{}
+	for _, peEdgeConfig := range edgeConfig.Spec.EdgePlaybook.Playbooks {
+		var playbookExec v1alpha1.PlaybookExecution
+		playbookExec.ObjectMeta.Name = edgeConfig.Name
+		playbookExec.ObjectMeta.OwnerReferences = []v1.OwnerReference{{
+			APIVersion: edgeConfig.APIVersion,
+			Kind:       edgeConfig.Kind,
+			Name:       edgeConfig.Name,
+			UID:        edgeConfig.UID,
+		}}
+		playbookExec.ObjectMeta.Namespace = edgeConfig.Namespace
+		playbookExec.Spec.Playbook = peEdgeConfig
+		playbookExec.Spec.ExecutionAttempt = 0
+		playbookExecutionStatus := v1alpha1.PlaybookExecutionStatus{
+			Conditions: []v1alpha1.PlaybookExecutionCondition{
+				{
+					Type:   v1alpha1.PlaybookExecutionDeploying,
+					Status: v1.ConditionTrue,
+				},
+			},
+		}
+
+		playbookExec.Status = playbookExecutionStatus
+		playbookExecutions = append(playbookExecutions, playbookExec)
+	}
+
+	return playbookExecutions
+}
+
+func (r *EdgeConfigReconciler) addPlaybookExecutionToDevices(ctx context.Context, edgeConfig *v1alpha1.EdgeConfig, edgeDevices []v1alpha1.EdgeDevice) error {
+	playbookExecutionBases := createPlaybookExecutions(edgeConfig)
```

Comment:
I'll return an error if len(edgeDevices) is 0 .

Reply:
Added in `Reconcile` method, not necessary anymore here (but thanks for the catch!)